client/src/legacy/entwine/LeftAndMain_Subsites.js

```diff
@@ -0,0 +1,190 @@
+          this.showReactiveNotice(subsiteInfo.subsiteName);
+        }
+      }, false);
+
+      // Dropdown change trigger
+      this.on('change', () => {
+        const subsiteID = this.val();
+        window.location.search = $.query.set('SubsiteID', subsiteID);
+      });
+
+      // We need to set when a page loads, as it may be e.g. the refresh of a currently blocked tab.
+      this.storeSubsiteInfo();
+    },
+    storeSubsiteInfo() {
+      const subsiteID = this.val();
+      const subsiteInfo = {
+        subsiteID,
+        subsiteName: $(`[value="${subsiteID}"]`, this).text()
+      };
+      window.localStorage.setItem('subsiteInfo', JSON.stringify(subsiteInfo));
+    },
+    showReactiveNotice(newSubsiteName) {
+      const { document } = window;
+      // React business
+      const modalContainer = this.getModalNode() || document.createElement('div');
+      document.body.appendChild(modalContainer);
+      const ChangeAlert = loadComponent('SubsiteChangeAlert');
+      const selectedNode = this.get(0);
+      const selectedIndex = selectedNode.selectedIndex;
+      ReactDom.render(
+        <ChangeAlert
+          otherTabSubsiteName={newSubsiteName}
+          myTabSubsiteName={selectedNode.options[selectedIndex].text}
+          myTabSubsiteID={this.val()}
+          revertCallback={changeActiveSubsite}
+        />,
+        modalContainer
+      );
+      this.setModalNode(modalContainer);
+    }
+  });
+
+  /*
+   * Reload subsites dropdown when links are processed
+   */
+  $('.cms-container .cms-menu-list li a').entwine({
+    onclick(e) {
+      $('.cms-container').loadFragment('admin/subsite_xhr', 'SubsiteList');
+      this._super(e);
+    }
+  });
+
+  /*
+   * Reload subsites dropdown when the admin area reloads (for deleting sites)
+   */
+  $('.cms-container .SubsiteAdmin .cms-edit-form fieldset.ss-gridfield').entwine({
+    onreload(e) {
+      $('.cms-container').loadFragment('admin/subsite_xhr', 'SubsiteList');
+      this._super(e);
+    }
+  });
+
+  /*
+   * Reload subsites dropdown when subsites are added or names are modified
+   */
+  $('.cms-container .tab.subsite-model').entwine({
+    onadd(e) {
+      $('.cms-container').loadFragment('admin/subsite_xhr', 'SubsiteList');
+      this._super(e);
+    }
+  });
+
+  // Subsite tab of Group editor
+  $('#Form_ItemEditForm_AccessAllSubsites').entwine({
+    /**
+     * Constructor: onmatch
+     */
+    onmatch() {
+      this.showHideSubsiteList();
+
+      const ref = this;
+      $('#Form_ItemEditForm_AccessAllSubsites input').change(() => {
+        ref.showHideSubsiteList();
+      });
+    },
+
+    showHideSubsiteList() {
+      $('#Form_ItemEditForm_Subsites').parent().parent().css('display', ($('#Form_ItemEditForm_AccessAllSubsites_1').is(':checked') ? 'none' : ''));
+    }
+  });
+
+  $('.cms-edit-form').entwine({
+    /**
+     * TODO: Fix with Entwine API extension. See https://github.com/silverstripe/silverstripe-subsites/pull/125
```

Comment:
is this TODO going to get done?

Reply:
No, this was existing code, moved in order to facilitate the new yarn build chain to support React components.
https://github.com/silverstripe/silverstripe-subsites/blob/fd10b868ec5dfccd8bd0273c92a30892e2af48cb/javascript/LeftAndMain_Subsites.js#L66 (current `2` branch HEAD)

Reply:
So, what you're saying is this TODO is just copy pasted from existing code

The link points to a PR merged in 2013, which is not really helpful

If so, just remove the TODO? It's currently just confusing